cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -332,6 +357,28 @@ func newMachineController(
 	}, nil
 }
 
+func getAPIGroupPreferredVersion(client discovery.DiscoveryInterface, APIGroup string) (string, error) {
+	groupList, err := client.ServerGroups()
+	if err != nil {
+		return "", fmt.Errorf("failed to get ServerGroups: %v", err)
+	}
+
+	foundAPIGroup := false
+	var APIVersion string
+	for _, group := range groupList.Groups {
+		if group.Name == APIGroup {
+			foundAPIGroup = true
+			APIVersion = group.PreferredVersion.Version
+			break
```

Comment:
Why not just `return` at this point? You can drop some variables and conditionals below.